sdk/attestation/azure-security-attestation/samples/basic-operations/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.13)
+
+project (attestation-basic-operations LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+macro (define_sample samplename)
+add_executable (
+  attestation-${samplename}
+  ${samplename}.cpp)
+
+create_per_service_target_build(attestation-${samplename} attestation)
+
+target_link_libraries(attestation-${samplename} PRIVATE azure-security-attestation azure-identity get-env-helper)
+
+# Add the sample to be run on CI. 
+# CI pipeline reads the {service}-samples.txt and runs the binaries listed there.
+file(APPEND ${CMAKE_BINARY_DIR}/attestation-samples.txt "${CMAKE_CURRENT_BINARY_DIR}/attestation-${samplename}\n")
+
+endmacro()
```

Comment:
I like your macro.
It'd be nice to have it as part of the `cmake-modules`. I might be a cool way to set the standard structure for a sample.

Let's keep it as a candidate for any future MQ sprint :)

Reply:
Yup.

I created it because I got annoyed that I kept forgetting to add the append to attestation-samples :).

It'll be trivial to adopt it to be more generic.